packages/xml-views-validation/src/validators/attributes/unknown-attribute-key.ts

```diff
@@ -0,0 +1,179 @@
+import { includes, find } from "lodash";
+import { XMLAttribute } from "@xml-tools/ast";
+import {
+  UI5SemanticModel,
+  UI5Aggregation,
+  UI5Association,
+  UI5Event,
+  UI5Prop,
+} from "@ui5-language-assistant/semantic-model-types";
+import {
+  getUI5ClassByXMLElement,
+  getUI5AggregationByXMLElement,
+  isXMLNamespaceKey,
+  flattenProperties,
+  flattenEvents,
+  flattenAssociations,
+  flattenAggregations,
+  ui5NodeToFQN,
+} from "@ui5-language-assistant/logic-utils";
+import { UnknownAttributeKeyIssue } from "../../../api";
+import { TEMPLATING_NS, CUSTOM_DATA_NS } from "../../utils/special-namespaces";
+
+export function validateUnknownAttributeKey(
+  attribute: XMLAttribute,
+  model: UI5SemanticModel
+): UnknownAttributeKeyIssue[] {
+  if (attribute.syntax.key === undefined || attribute.key === null) {
+    // Can't give an error without a position or value
+    return [];
+  }
+
+  if (isAttributeNameAlwaysValid(attribute)) {
+    return [];
+  }
+
+  // Only validate (known) classes and aggregations
+  const xmlElement = attribute.parent;
+  const ui5Class = getUI5ClassByXMLElement(xmlElement, model);
```

Comment:
Could ` const ui5Class = getUI5ClassByXMLElement(xmlElement, model);` be moved to line 58?

Reply:
It would work (since aggregations and classes names don't intersect) but the check for class is more straight-forward (it only checks the current tag) so I think it should be done first. (It's also slightly more efficient since we don't go over all the aggregations of the parent class, but I don't think that matters much since there aren't many aggregations for each class.)